### DISCUSSION TITLE
encountered an I/O error: encoding has unexpected compression mode

### DISCUSSION THREAD

**Discussion by Meest-NAM** [filecoin-project/lotus#7949] [Storage Provider]
I have window post problem
Generate_Window_Post finished but runPost failed
How can I solve ??
Daemon:  1.6.0+mainnet+git.3fc23a785+api1.0.1
Local: lotus-miner version 1.6.0+mainnet+git.3fc23a785
CPU : AMD3970x
RAM : 256G
GPU : RTX 3080
Nvme(m.2) : 500G (OS and params)
Nvme(m.2) : 2.0TB (.lotus and .lotusminer)

2021-04-22T07:25:22.803 INFO filcrypto::proofs::api > generate_window_post: finish
2021-04-22T07:25:22.805+0900 [34mINFO [0m storageminer storage/wdpost_run.go:600 computing window post {"batch": 0, "elapsed": 469.647362632}
2021-04-22T07:25:22.805+0900 [31mERROR [0m storageminer storage/wdpost_run.go:104 runPost failed: running window post failed:
   github.com/filecoin-project/lotus/storage.(*WindowPoStScheduler).runPost
    /root/lotus/storage/wdpost_run.go:650
 - encountered an I/O error: encoding has unexpected compression mode
  
  Caused by:
    encoding has unexpected compression mode
  
  Stack backtrace:
    0: storage_proofs_core::compound_proof::CompoundProof::circuit_proofs
    1: storage_proofs_core::compound_proof::CompoundProof::prove
    2: filecoin_proofs::api::post::generate_window_post
    3: filecoin_proofs_api::post::generate_window_post_inner
    4: filecoin_proofs_api::post::generate_window_post
    5: <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
    6: ffi_toolkit::catch_panic_response
    7: fil_generate_window_post
    8: _cgo_a8fa62747d41_Cfunc_fil_generate_window_post
         at /tmp/go-build/cgo-gcc-prolog:596
    9: runtime.asmcgocall
         at /usr/local/go/src/runtime/asm_amd64.s:656
   github.com/filecoin-project/filecoin-ffi.GenerateWindowPoSt
  /root/lotus/extern/filecoin-ffi/proofs.go:587
   github.com/filecoin-project/lotus/extern/sector-storage/ffiwrapper.(*Sealer).GenerateWindowPoSt
  /root/lotus/extern/sector-storage/ffiwrapper/verifier_cgo.go:45
   github.com/filecoin-project/lotus/storage.(*WindowPoStScheduler).runPost
  /root/lotus/storage/wdpost_run.go:597
   github.com/filecoin-project/lotus/storage.(*WindowPoStScheduler).runGeneratePoST
  /root/lotus/storage/wdpost_run.go:102
   github.com/filecoin-project/lotus/storage.(*WindowPoStScheduler).startGeneratePoST.func1
  /root/lotus/storage/wdpost_run.go:86
  runtime.goexit
  /usr/local/go/src/runtime/asm_amd64.s:1374
2021-04-22T07:25:22.805+0900 [31mERROR [0m storageminer storage/wdpost_run.go:48 Got err running window post failed:
   github.com/filecoin-project/lotus/storage.(*WindowPoStScheduler).runPost
    /root/lotus/storage/wdpost_run.go:650
 - encountered an I/O error: encoding has unexpected compression mode
  
  Caused by:
    encoding has unexpected compression mode
  
  Stack backtrace:
    0: storage_proofs_core::compound_proof::CompoundProof::circuit_proofs
    1: storage_proofs_core::compound_proof::CompoundProof::prove
    2: filecoin_proofs::api::post::generate_window_post
    3: filecoin_proofs_api::post::generate_window_post_inner
    4: filecoin_proofs_api::post::generate_window_post
    5: <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
    6: ffi_toolkit::catch_panic_response
    7: fil_generate_window_post
    8: _cgo_a8fa62747d41_Cfunc_fil_generate_window_post
         at /tmp/go-build/cgo-gcc-prolog:596
    9: runtime.asmcgocall
         at /usr/local/go/src/runtime/asm_amd64.s:656
   github.com/filecoin-project/filecoin-ffi.GenerateWindowPoSt
  /root/lotus/extern/filecoin-ffi/proofs.go:587
   github.com/filecoin-project/lotus/extern/sector-storage/ffiwrapper.(*Sealer).GenerateWindowPoSt
  /root/lotus/extern/sector-storage/ffiwrapper/verifier_cgo.go:45
   github.com/filecoin-project/lotus/storage.(*WindowPoStScheduler).runPost
  /root/lotus/storage/wdpost_run.go:597
   github.com/filecoin-project/lotus/storage.(*WindowPoStScheduler).runGeneratePoST
  /root/lotus/storage/wdpost_run.go:102
   github.com/filecoin-project/lotus/storage.(*WindowPoStScheduler).startGeneratePoST.func1
  /root/lotus/storage/wdpost_run.go:86
  runtime.goexit
  /usr/local/go/src/runtime/asm_amd64.s:1374 - TODO handle errors
2021-04-22T07:25:22.806+0900 [33mWARN [0m storageminer storage/wdpost_changehandler.go:252 Aborted window post Proving (Deadline: &{CurrentEpoch:691235 PeriodStart:691245 Index:0 Open:691245 Close:691305 Challenge:691225 FaultCutoff:691175 WPoStPeriodDeadlines:48 WPoStProvingPeriod:2880 WPoStChallengeWindow:60 WPoStChallengeLookback:20 FaultDeclarationCutoff:70}) (edited)

**Comment by Reiers** [comment]
**Hi @namoo-cloud** 

Thanks for the report. The issue you are experiencing lies elsewhere.

Moving this issue to Lotus Discussions for help and troubleshooting.

Closing Lotus issue ticket.

**Comment by LexLuthr** [comment]
@namoo-cloud Do you still need help with this problem? If not, then can you please tell us how you fixed it?

**Reply by LexLuthr** [reply]
I am archiving this discussion as there was no update from the author.
